src/commands/parser.rs

```diff
@@ -1,9 +1,196 @@
-use nom::{*, bytes::complete::{take_until, take_till}, character::{is_newline}, error::*, sequence::*};
+use nom::{
+    bytes::complete::{take_until, tag},
+    character::complete::{newline, not_line_ending},
+    combinator::opt,
+    multi::many0,
+    *,
+};
 
-pub fn todo() -> impl Fn(&[u8]) -> IResult<&[u8], &[u8], Error<&[u8]>> {
-    move |i: &[u8]| {
-	let header = take_until("TODO:");
-	let content = take_till(is_newline);
-	preceded(header, content)(i)
+#[derive(Debug, PartialEq, Clone)]
+pub struct Todo {
+    pub tag: String,
+    pub text: String,
+}
```

Comment:
Maybe we should not restrain ourselves to the name `Todo` given how custom we aim this to be. Do you think it is better to generalize it later on or now?

Reply:
Todo is a metaphysical concept not attached to an actual todo.